src/lib/sd2/fswatch_connections.py

```diff
@@ -156,33 +164,32 @@ def __init__(self, args, workspaces):
             logging.info("%s %s", wi['name'], strcmd)
             wi['fswatchproc'] = subprocess.Popen(
                 strcmd,
-                shell=True,
-                bufsize=1,  # line buffered
+                shell=True, # with this option the pid returned is of the shell: https://docs.python.org/2/library/subprocess.html#subprocess.Popen.pid
+                bufsize=1, # line buffered
                 stdout=subprocess.PIPE,
                 # stderr=subprocess.PIPE,
                 close_fds=ON_POSIX
             )
             fl = fcntl.fcntl(wi['fswatchproc'].stdout, fcntl.F_GETFL)
-            fcntl.fcntl(wi['fswatchproc'].stdout, fcntl.F_SETFL,
-                        fl | os.O_NONBLOCK)
+            fcntl.fcntl(wi['fswatchproc'].stdout, fcntl.F_SETFL, fl | os.O_NONBLOCK)
         logging.debug("FSW:LL")
 
     def poll(self):
         for ws in self._workspaces:
             proc = ws['fswatchproc']
             try:
-                line = proc.stdout.readline().strip()
+                line = proc.stdout.readline()
```

Comment:
I am not sure what we accomplish here. Is there any other code that can generate IOError?

Reply:
I am not 100% sure either but after a lot of attempts and research this is the only thing that stopped the "Resource not yet available" error

Reply:
I can see why the .strip() might have something to do and I am fine with that, but you moved a lot of code inside the exception block, and cannot see why this is necessary

Reply:
As they are all part of the same logic I though it best to group them up. Reverted anyway